CMakeLists.txt

```
@@ -40,64 +44,77 @@ execute_process(COMMAND ${BASH} ${PROJECT_SOURCE_DIR}/scripts/gen-header-list.sh
   )
 include(${PROJECT_BINARY_DIR}/camp_headers.cmake)
 
+# backends
 set (camp_depends)
 
-if (ENABLE_OPENMP)
-  set (camp_depends
-    openmp)
-endif()
+set (camp_backends openmp cuda hip sycl)
+# NOTE: camp itself doesn't require a hip or cuda compiler,
+#       but some of its features are activated by having a device compiler
+set (camp_runtime_backends cuda hip)
+
+foreach (backend ${camp_backends})
+  string(TOUPPER "${backend}" suffix)
+  if ("${ENABLE_${suffix}}")
+    set ("CAMP_ENABLE_${suffix}" On)
+    if (backend IN_LIST camp_runtime_backends)
+      list (APPEND camp_depends ${backend}_runtime)
+    else()
+      list (APPEND camp_depends ${backend})
+    endif()
+  endif()
+endforeach()
 
-if(ENABLE_CUDA)
+if (ENABLE_CUDA)
   if("${CUDA_VERSION_STRING}" VERSION_LESS "9.2")
     message(FATAL_ERROR "Trying to use CUDA version ${CUDA_VERSION_STRING}. CAMP dependency Googletest requires CUDA version 9.2.x or newer.")
```

Making error message correspond to new version.
```suggestion
    message(FATAL_ERROR "Trying to use CUDA version ${CUDA_VERSION_STRING}. CAMP dependency Googletest requires CUDA version 10.1.x or newer.")
```

---

Good point, I need to change the message a bit more though, since this is now our requirement exceeding the requirement from google test.  Thanks @rchen20.

```suggestion
    message(FATAL_ERROR "Trying to use CUDA version ${CUDA_VERSION_STRING}. CAMP requires CUDA version 10.1 or newer.")
```